atc/engine/builder/builder.go

```diff
@@ -63,185 +64,185 @@ func (builder *stepBuilder) BuildStep(build db.Build) (exec.Step, error) {
 		return exec.IdentityStep{}, errors.New("Schema not supported")
 	}
 
-	return builder.buildStep(build, build.PrivatePlan()), nil
+	return builder.buildStep(build, build.PrivatePlan(), credVarsTracker), nil
```

Comment:
(Follow-up from previous comment:) this is where it could construct the `credVarsTracker` itself if it had the secrets

Reply:
Moved `credVarsTracker` construction to `stepBuilder`.